docs/metrics/iou.md

```diff
@@ -0,0 +1,170 @@
+# Intersection over Union (IoU)
+
+Intersection over Union (IoU) measures the ratio of the intersection and the union between two instances, ranging from 0 to 1 where 1 indicates a perfect match.  It is one of the metrics that measure the similarity between two instances, often used to compare predictions to ground truths.
```

Comment:
> two instances

Two instances of what?

Reply:
I understand the desire to be 100% correct here, and that IoU isn't _always_ for geometry, but it's not exactly clear what this paragraph means without mentioning geometry.